core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java

```diff
@@ -47,16 +36,16 @@ public RegexpQueryParser() {
     }
 
     @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
         String fieldName = parser.currentName();
-        String rewriteMethod = null;
+        String rewrite = null;
 
-        Object value = null;
+        String value = null;
```

Comment:
we shouldn't make this change here in the branch, but rather in master. Let's get that one in first in master and then this one in our branch.